homeassistant/core.py

```diff
@@ -1669,6 +1670,15 @@ def __init__(
         self.context = context or Context()
         self.state_info = state_info
         self.domain, self.object_id = split_entity_id(self.entity_id)
+        # last_updated_timestamp will nearly always be called by
+        # the recorder or websocket_api so we do not need to
+        # generate it lazily.
+        if last_updated_timestamp is not None:
+            # We round to 6 decimal places to match the precision
+            # of .timestamp()
+            self.last_updated_timestamp = round(last_updated_timestamp, 6)
+        else:
+            self.last_updated_timestamp = self.last_updated.timestamp()
```

Comment:
Wasn't expecting to have to `round()` here, need to profile `.timestamp()` vs `round()`